core/http-auth-spi/src/main/java/software/amazon/awssdk/http/auth/spi/signer/SignerProperty.java

```diff
@@ -52,6 +58,18 @@ public static <T> SignerProperty<T> create(Class<?> namespace, String name) {
         return new SignerProperty<>(namespace.getName(), name);
     }
 
+    private void ensureUnique() {
```

Comment:
Can we add some javadocs to the `create()` method that tells the reader about the uniqueness constraint? That'll also help clarify more that `Class<?> namespace` is not for the type of property value. See earlier discussion in https://github.com/aws/aws-sdk-java-v2/pull/4448#discussion_r1330557121

Reply:
Fixed